Sources_v3/StreamChatUI/ChatChannel/ChatChannelMessageInputTextView.swift

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+open class ChatChannelMessageInputTextView<ExtraData: UIExtraDataTypes>: UITextView,
+    AppearanceSetting,
+    Customizable,
+    UIConfigProvider
+{
+    // MARK: - Subviews
+    
+    public lazy var placeholderLabel: UILabel = UILabel().withoutAutoresizingMaskConstraints
+    
+    // MARK: - Overrides
+    
+    override public var text: String! {
+        didSet {
+            textDidChange()
+        }
+    }
+    
+    override public var attributedText: NSAttributedString! {
+        didSet {
+            textDidChange()
+        }
+    }
+    
+    override open func didMoveToSuperview() {
+        super.didMoveToSuperview()
+        guard superview != nil else { return }
+        
+        setUp()
+        (self as! Self).applyDefaultAppearance()
+        setUpAppearance()
+        setUpLayout()
+        updateContent()
+    }
+    
+    // MARK: Public
+    
+    open func defaultAppearance() {
+        font = .preferredFont(forTextStyle: .callout)
+        textContainer.lineFragmentPadding = 10
```

Comment:
Is it a good idea to introduce additional inner padding of the text? Wouldn't it be better to set this to `0` a position the view correctly within its superview?

Reply:
Actually we are not introducing it.
The default value was `5` and we need to adjust it just because of `cornerRadius` so it will look better.
IMO it's the right property to play with in this case.